### PR TITLE
Calculator: Visual feedback on keyboard input

### PR DESCRIPTION
**Pull request by ForLoveOfCats**
https://user-images.githubusercontent.com/30666851/152081816-30449a58-6d95-42d1-935b-5097eed02129.mp4

Adds the concept of a button being "ghosted" where it is painted as though it is being clicked regardless of if it actually is. This is used by Calculator to provide visual feedback on keyboard input.

I'm not the fondest of the term "ghosted" so suggestions in that regard are welcome.

All keypad buttons apart from the equal button have been prevented from gaining focus. This means that when the keypad is focused the return key will always perform the evaluation action instead of clicking a different focused key.

To augment these changes more calculator actions have had keyboard input support added. Every button apart from memory and clear error now have corresponding keyboard inputs. The keys used for square root and inverse are probably not ideal (currently `s` and `i` respectively).

### CODE REVIEW

**Comment by bgianfo** [comment]
Instead of ghost, how about:
- mimic
- impersonate
- imitate

**Comment by ForLoveOfCats** [comment]
Oh I like `mimic_pressed` :+1:

**Comment by AtkinsSJ** [comment]
Hi, this has a merge conflict that needs resolving.

**Comment by ForLoveOfCats** [comment]
Rebased :)

File: Userland/Applications/Calculator/CalculatorWidget.h [hunk 1]
```diff
@@ -54,4 +56,6 @@ class CalculatorWidget final : public GUI::Widget {
     RefPtr<GUI::Button> m_inverse_button;
     RefPtr<GUI::Button> m_percent_button;
     RefPtr<GUI::Button> m_equals_button;
+
+    Optional<RefPtr<GUI::Button>> m_mimic_pressed_button {};
```
Comment (awesomekling) [inline Userland/Applications/Calculator/CalculatorWidget.h:61]: `RefPtr` can be null, so I don't see a need to wrap it in `Optional` :)

**Pull request merged**